### PR TITLE
Support MySQL field filter and mongo collection show

### PR DESCRIPTION
**Pull request by HaitaoDeng**
image
image
image

### CODE REVIEW

File: lib/api/mongo/mongo_tables_api.dart [hunk 1]
```diff
@@ -30,4 +33,34 @@ class MongoTablesApi {
...
     }).toList();
   }
+
+  static Future<MongoSqlResult> getTableData(
+      String addr, String username, String password, String databaseName, String tableName) async {
+    var db = await Db.create(addr + "/" + databaseName);
+    await db.open();
+    var collection = await db.collection(tableName);
+    SelectorBuilder builder = SelectorBuilder().limit(100);
+    // Stream<Map<String, dynamic>> data = await collection.find(builder);
```
Comment (goflutterjava) [inline lib/api/mongo/mongo_tables_api.dart:43]: commented out code, need delete

File: lib/api/mysql/mysql_databases_api.dart [hunk 2]
```diff
@@ -17,15 +17,18 @@
 // under the License.
 //
 
+import 'dart:developer';
```
Comment (goflutterjava) [inline lib/api/mysql/mysql_databases_api.dart:20]: why use `'dart:developer'` package

**Review by goflutterjava — CHANGES_REQUESTED** [review]

File: lib/api/mongo/mongo_tables_api.dart [hunk 3]
```diff
@@ -30,4 +33,34 @@ class MongoTablesApi {
       return new TableResp(name);
     }).toList();
   }
+
+  static Future<MongoSqlResult> getTableData(
+      String addr, String username, String password, String databaseName, String tableName) async {
+    var db = await Db.create(addr + "/" + databaseName);
```
Comment (Shoothzj) [inline lib/api/mongo/mongo_tables_api.dart:39]: potential memory leak here, db need close

File: lib/ui/component/dynamic_filter_table.dart [hunk 4]
```diff
@@ -0,0 +1,229 @@
...
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+//
+
+import 'package:flutter/material.dart';
+import 'package:paas_dashboard_flutter/generated/l10n.dart';
+import 'package:paas_dashboard_flutter/vm/mysql/mysql_table_data_view_model.dart';
+
+typedef VoidCallback = void Function();
```
Comment (Shoothzj) [inline lib/ui/component/dynamic_filter_table.dart:24]: dart has a `VoidCallback`, no need to define it

File: lib/api/mongo/mongo_tables_api.dart (hunk 1 above)
Comment (HaitaoDeng) [reply lib/api/mongo/mongo_tables_api.dart]: OK

File: lib/ui/component/dynamic_filter_table.dart (hunk 4 above)
Comment (HaitaoDeng) [reply lib/ui/component/dynamic_filter_table.dart]: OK

File: lib/ui/mongo/widget/mongo_table_data.dart [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+//
+
+import 'package:flutter/cupertino.dart';
```
[the remark this answers is not in the document]
Comment (HaitaoDeng) [reply lib/ui/mongo/widget/mongo_table_data.dart]: en

**Review by Shoothzj — APPROVED** [review]

**Review by goflutterjava — APPROVED** [review]

**Pull request merged**